etcd3/client.py

```diff
@@ -126,6 +174,131 @@ def _get_secure_creds(self, ca_cert, cert_key, cert_cert):
                         cert_cert_file.read()
                     )
 
+    def _discover(self, srv_name):
+        answers = dns.resolver.query(srv_name, 'SRV')
+        hosts = {}
+        for answer in answers:
+            node = Node(
+                answer.target.to_text(omit_final_dot=True),
+                answer.port,
+                self.uses_secure_channel
+            )
+            hosts[node.netloc] = node
+        if not len(hosts):
+            raise exceptions.NoServerAvailableError(
+                "No host were found in the SRV query for {name}".format(
+                    name=srv_name
+                )
+            )
+        return hosts
+
+    @property
+    def current_node(self):
+        if self._current_node is None:
+            return None
+        return self._nodes_cache[self._current_node]
+
+    @property
+    def channel(self):
+        """
+        Get an available channel on the first node that is not failed.
+
+        Raises an exception if no node is available.
+        """
+        # No reconnection is allowed, just use the current node.
+        if not self.reconnect:
+            return self.current_node.channel
+
+        # If we already picked a node and it didn't fail,
+        # return its channel
+        if self._current_node and not self.current_node.failed:
+            return self.current_node.use()
+        else:
+            available = [m for m in self._nodes_cache.values()
+                         if not m.failed]
+            for node in available:
+                # A node might have failed in the meanwhile
+                if node.failed:
+                    next
```

Comment:
I don't think this does what you think it does

```
>>> next
<built-in function next>
```

You meant `continue` I imagine